### PR TITLE
[Import CSV] Improvements to importCsv Cloud Function

### PR DESCRIPTION
**Pull request by gino-m**
Contributes to #466. Specifically:

- [x] Resolves CORS issues encountered during development.
- [x] Adds aliases for fields recognized by app, including location, caption, and id.
- [x] Treats all other columns as custom key-value pairs stored in feature's "attributes" map.

@Tiyarah could you PTAL?

### CODE REVIEW

**Comment by gino-m** [comment]
Failing tests seem to be unrelated to this change. Working on a fix for those now.

File: functions/common/datastore.js [hunk 1]
```diff
@@ -66,28 +68,22 @@ class Datastore {
...
-      caption: caption,
-      location: location
-    });
+    await docRef.collection("features").add(feature);
   }
 }
 
-module.exports = Datastore;
\ No newline at end of file
+module.exports = Datastore;
```
Comment (Tiyarah) [inline functions/common/datastore.js:89]: This is identical as far as I see!

**Review by Tiyarah — APPROVED** [review]

File: functions/package.json [hunk 2]
```diff
@@ -19,7 +20,7 @@
     "requests": "^0.3.0"
   },
   "engines": {
-    "node": "13"
+    "node": "12"
```
[the remark this answers is not in the document]
Comment (gino-m) [reply functions/package.json:23]: Yes, realized late that according to the docs only 8, 10, and 12 are currently supported.
https://firebase.google.com/docs/functions/get-started#set-up-node.js-and-the-firebase-cli

File: functions/import-csv.js [hunk 3]
```diff
@@ -15,58 +15,136 @@
...
+}
+
+/**
+ * Convert the provided row (key-value pairs) and layerId into a Feature for
+ * insertion into the data store.
+ */
+function csvRowToFeature(row, layerId) {
+  let data = { layerId };
+  let attributes = {};
+  for (const key in row) {
```
[the remark this answers is not in the document]
Comment (gino-m) [reply functions/import-csv.js]: Yep! How about `columnName`?

File: functions/import-csv.js [hunk 4]
```diff
@@ -15,58 +15,136 @@
...
+const SPECIAL_COLUMN_NAMES = invertAndFlatten({
+  id: ["id", "key"],
+  caption: ["caption", "name", "label"],
+  lat: ["lat", "latitude", "y"],
+  lng: ["lng", "lon", "long", "lng", "x"],
+});
+
+async function insertRow(projectId, layerId, record) {
+  const feature = csvRowToFeature(record, layerId);
+  if (feature) {
```
[the remark this answers is not in the document]
Comment (gino-m) [reply functions/import-csv.js:116]: Good catch, thanks!

File: functions/common/datastore.js (hunk 1 above)
Comment (gino-m) [reply functions/common/datastore.js:89]: There were a few changes to `insertFeature()`. Sorry for the formatting change!

**Review by gino-m — COMMENTED** [review]
Thanks for the comments! RFAL 👀

**Review by Tiyarah — APPROVED** [review]

**Pull request merged**